### PR TITLE
Feature/init services by priority

### PR DESCRIPTION
**Pull request by TroubleMakerBen**
* I have renamed `MSPriorityHigh` to `MSPriorityMax` to indicate it's the MAX priority. If we need a `MSPriorityHigh` we can introduce this at a later point.
* `MSPriority` has to be ordered, otherwise sorting won't work.
* Modules are sorted based on their priority and then initialized.

### CODE REVIEW

**Comment by jaelim-ms** [comment]
It shouldn't be coupled with channel priority. Initialize order and channel priority are completely different I think.

File: MobileCenter/MobileCenter/MSMobileCenter.m [hunk 1]
```diff
@@ -176,12 +176,36 @@ - (BOOL)configure:(NSString *)appSecret {
 - (void)start:(NSString *)appSecret withServices:(NSArray<Class> *)services {
   BOOL configured = [self configure:appSecret];
   if (configured) {
-    for (Class service in services) {
+
+    NSArray *sortedServices = [self sortServices:services];
+
+
```
Comment (clpolet) [inline MobileCenter/MobileCenter/MSMobileCenter.m]: Too much space here :o)

**Comment by TroubleMakerBen** [comment]
I have addressed @jaelim-ms 's comment and introduced a new property and added tests for that.
There's also 1 additional check in MSCrashes.mm for an empty data object now that I noticed when I ran our puppet app.

File: MobileCenter/MobileCenter/MSMobileCenter.m [hunk 2]
```diff
@@ -176,12 +176,35 @@ - (BOOL)configure:(NSString *)appSecret {
...
+    NSArray *sortedServices = [self sortServices:services];
+
+    for (Class service in sortedServices) {
       [self startService:service];
     }
   }
 }
 
+- (NSArray *)sortServices:(NSArray<Class> *)services {
+  // Sort services in descending order to make sure the service with the highest priority gets initialized first.
```
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/MSMobileCenter.m]: new line. ⚠️

File: MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm [hunk 3]
```diff
@@ -506,17 +510,18 @@ - (void)processLogBufferAfterCrash {
...
+      if(serializedLog && serializedLog.length && serializedLog.length > 0) {
+        id <MSLog> item = [NSKeyedUnarchiver unarchiveObjectWithData:serializedLog];
+        if (item) {
+          if ([((NSObject *) item) isKindOfClass:[MSAppleErrorLog class]]) {
+            [self.logManager processLog:item withPriority:self.priority];
+          } else {
+            [self.logManager processLog:item withPriority:MSPriorityDefault];
+          }
         }
+        // Create empty new file, overwrites the old one.
```
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm]: new line

**Pull request merged**